Assets/Scripts/NPC/WalkingState.cs

```diff
@@ -0,0 +1,82 @@
+using UnityEngine;
+
+public class WalkingState : SmallEnemiesBaseState
+{
+    [SerializeField] private Waypoint currentWaypoint;
+    [SerializeField] private Waypoint lastWaypoint;
+    [SerializeField] private Grid myGrid;
+    [SerializeField] private float distanceTreshhold;
+    [SerializeField] private float movingSpeed;
+
+    [Header("Performance")]
+    [SerializeField] private int maxWaypointCallStack = 2;
+
+    private int newWaypointCallStack;
+    private Rigidbody2D _rigidbody;
+    private Vector2 _moveVelocity;
```

Comment:
```suggestion
    [Header("Grid")]
    [SerializeField] private Waypoint currentWaypoint;
    [SerializeField] private Grid parentGrid; // gridSystem kan ook
    
    [Header("Settings")]
    [SerializeField] private float distanceTreshhold;
    [SerializeField] private float movingSpeed;

    [Header("Performance")]
    [SerializeField] private int maxWaypointCallStack = 2;

    private int newWaypointCallStack;
    private Rigidbody2D _rigidbody;
    private Vector2 _moveVelocity;
    private Waypoint _lastWaypoint;
```
lastWaypoint is niet nodig om in de inspector te veranderen. Daar naast een betere naam voor "myGrid". Ook wat header toegevoegd want je had er 1 ipv alles een header geven.